infrastructure/src/main/java/org/corfudb/infrastructure/CompactorLeaderServices.java

```diff
@@ -0,0 +1,480 @@
+            finishCompactionCycle();
+        } else if (currentTime - LivenessValidatorHelper.getPrevActiveTime() > timeout &&
+                managerStatus != null && managerStatus.getStatus() == StatusType.STARTED) {
+            syslog.info("No active client checkpointers available...");
+            try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                managerStatus = (CheckpointingStatus) txn.getRecord(
+                        DistributedCompactor.COMPACTION_MANAGER_TABLE_NAME,
+                        DistributedCompactor.COMPACTION_MANAGER_KEY).getPayload();
+
+                TransactionalContext.getCurrentContext().setPriorityLevel(CorfuMessage.PriorityLevel.HIGH);
+                txn.putRecord(compactionManagerTable,
+                        DistributedCompactor.COMPACTION_MANAGER_KEY,
+                        buildCheckpointStatus(StatusType.STARTED_ALL,
+                                managerStatus.getTableSize(),
+                                managerStatus.getTimeTaken()),
+                        null);
+                txn.commit();
+                LivenessValidatorHelper.clear();
+            } catch (Exception e) {
+                syslog.warn("Exception in handleNoActiveCheckpointers, e: {}. StackTrace: {}", e, e.getStackTrace());
+            }
+        }
+    }
+
+    private boolean handleSlowCheckpointers(TableName table) {
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            CheckpointingStatus tableStatus = txn.getRecord(
+                    checkpointingStatusTable, table).getPayload();
+            if (tableStatus.getStatus() != StatusType.COMPLETED && tableStatus.getStatus() != StatusType.FAILED) {
+                TransactionalContext.getCurrentContext().setPriorityLevel(CorfuMessage.PriorityLevel.HIGH);
+                txn.putRecord(checkpointingStatusTable,
+                        table,
+                        buildCheckpointStatus(StatusType.FAILED),
+                        null);
+                txn.delete(activeCheckpointsTable, table);
+                //Mark cycle failed and return on first failure
+                txn.putRecord(compactionManagerTable,
+                        DistributedCompactor.COMPACTION_MANAGER_KEY,
+                        buildCheckpointStatus(StatusType.FAILED),
+                        null);
+                txn.commit();
+                syslog.warn("Finished compaction cycle. FAILED due to table: {}${}", table.getNamespace(),
+                        table.getTableName());
+                return false;
+            } else {
+                txn.delete(activeCheckpointsTable, table);
+                txn.commit();
+            }
+        } catch (TransactionAbortedException ex) {
+            if (ex.getAbortCause() == AbortCause.CONFLICT) {
+                syslog.warn("Another node tried to commit");
+                readCache.clear();
+                return true;
```

Comment:
do we return true because another node was able to resume checkpointing?  What do the return types indicate here?

Reply:
Here the return type indicates whether the compaction has to proceed or not, ie.
if we decide to mark a table as FAILED due to timeout, then we do not proceed with CPing further tables. 

So at line 339, it returns true because we can continue with CP as the slow checkpointer seems to have updated the table's CP status.

Reply:
Please return enum then,  and add your explanation into it